Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -461,20 +467,23 @@ BTDiagnostics::InitializeBufferData ( int i_buffer , int lev)
     // number of cells in z is modified since each buffer multifab always
     // contains a minimum m_buffer_size=256 cells
     int num_z_cells_in_snapshot = m_max_buffer_multifabs[i_buffer] * m_buffer_size;
-    m_snapshot_domain_lab[i_buffer] = diag_dom;
-    m_snapshot_domain_lab[i_buffer].setLo(m_moving_window_dir,
-                                  zmin_buffer_lab + warpx.moving_window_v * m_t_lab[i_buffer]);
-    m_snapshot_domain_lab[i_buffer].setHi(m_moving_window_dir,
-                                  zmax_buffer_lab + warpx.moving_window_v * m_t_lab[i_buffer]);
-    // To prevent round off errors, moving the snapshot domain by half a cell so that all the slices
-    // lie close to the cell-centers in the lab-frame grid instead of on the edge of cell.
-    amrex::Real new_hi = m_snapshot_domain_lab[i_buffer].hi(m_moving_window_dir)
-                       + 0.5_rt * dz_lab(warpx.getdt(lev), ref_ratio[m_moving_window_dir]);
-    m_snapshot_domain_lab[i_buffer].setHi(m_moving_window_dir,new_hi);
-    amrex::Real new_lo = m_snapshot_domain_lab[i_buffer].hi(m_moving_window_dir) -
-                         num_z_cells_in_snapshot *
-                         dz_lab(warpx.getdt(lev), ref_ratio[m_moving_window_dir]);
-    m_snapshot_domain_lab[i_buffer].setLo(m_moving_window_dir, new_lo);
+    if (restart == false) {
+        m_snapshot_domain_lab[i_buffer] = diag_dom;
+        m_snapshot_domain_lab[i_buffer].setLo(m_moving_window_dir,
+                                      zmin_buffer_lab + warpx.moving_window_v * m_t_lab[i_buffer]);
+        m_snapshot_domain_lab[i_buffer].setHi(m_moving_window_dir,
+                                      zmax_buffer_lab + warpx.moving_window_v * m_t_lab[i_buffer]);
+        // To prevent round off errors, moving the snapshot domain by half a cell so that all the slices
+        // lie close to the cell-centers in the lab-frame grid instead of on the edge of cell.
+        amrex::Real new_hi = m_snapshot_domain_lab[i_buffer].hi(m_moving_window_dir)
+                           + 0.5_rt * dz_lab(warpx.getdt(lev), ref_ratio[m_moving_window_dir]);
+        m_snapshot_domain_lab[i_buffer].setHi(m_moving_window_dir,new_hi);
+        amrex::Real new_lo = m_snapshot_domain_lab[i_buffer].hi(m_moving_window_dir) -
+                             num_z_cells_in_snapshot *
+                             dz_lab(warpx.getdt(lev), ref_ratio[m_moving_window_dir]);
+        m_snapshot_domain_lab[i_buffer].setLo(m_moving_window_dir, new_lo);
+    } else {
+    }
```

Comment:
As noted by CodeQL, this `else` block is empty. Do we need it? Or can we just keep the first `if` block alone?

Reply:
fixed. Thank you